src/hooks/use-imperative-dialog.tsx

```diff
@@ -51,13 +74,17 @@ type ImperativeDialogProps = {
   text: string
   confirmText?: string
   cancelText?: string
+  extraConfirmation?: boolean
+  entityName?: string
 }
```

Comment:
nit(non-blocking): There is an edge case where if no `entityName` is passed but `extraConfirmation` is true it's possible to just confirm the modal by default: 
![image](https://user-images.githubusercontent.com/88927411/210613257-17442ef8-4d86-4a8e-8bf6-a4aa0ac2656e.png)

it's possible to prevent by a bit of typing like this: 
```
type ImperativeDialogProps =
  | {
      heading: string
      text: string
      confirmText?: string
      cancelText?: string
    } & (
      | {
          extraConfirmation: true
          entityName: string
        }
      | {
          extraConfirmation: undefined
          entityName: undefined
        }
    )

```

Reply:
Ah neat, decent way of typing a required-together type relationship without a sub-object. Will update.

Reply:
Done.